typescript/infra/scripts/funding/fund-keys-from-deployer.ts

```diff
@@ -417,29 +422,47 @@ class ContextFunder {
     );
   }
 
-  // The keys here are not ReadOnlyCloudAgentKeys, instead they are AgentGCPKey or AgentAWSKeys,
-  // which require credentials to fetch. If you want to avoid requiring credentials, use
-  // fromSerializedAddressFile instead.
-  static async fromContext(
+  // the keys are retrived from the local artifacts in the infra/config/relayer.json or infra/config/kathy.json
+  static async fromLocal(
     environment: DeployEnvironment,
     multiProvider: MultiProvider,
     context: Contexts,
-    rolesToFund: Role[],
+    rolesToFund: FundableRole[],
     skipIgpClaim: boolean,
   ) {
-    const agentConfig = getAgentConfig(context, environment);
-    const roleKeysPerChain = getRoleKeysPerChain(agentConfig);
-    // Fetch all the keys
-    await promiseObjAll(
-      objMap(roleKeysPerChain, (_chain, roleKeys) => {
-        return promiseObjAll(
-          objMap(roleKeys, (_role, keys) => {
-            return Promise.all(keys.map((key) => key.fetch()));
-          }),
+    // only roles that are fundable keys ie. relayer and kathy
+    const fundableRoleKeys: Record<FundableRole, Address> = {
+      [Role.Relayer]: '',
+      [Role.Kathy]: '',
+    };
+    for (const role of rolesToFund) {
+      assertFundableRole(role); // only the relayer and kathy are fundable keys
+      const roleAddress = fetchLocalKeyAddresses(role)[environment][context];
+      if (!roleAddress) {
+        throw Error(
+          `Could not find address for ${role} in ${environment} ${context}`,
         );
-      }),
-    );
-
+      }
+      fundableRoleKeys[role] = roleAddress;
+    }
+    const chains = getEnvironmentConfig(environment).chainMetadataConfigs;
+    const roleKeysPerChain: ChainMap<Record<FundableRole, BaseAgentKey[]>> = {};
+    for (const chain of Object.keys(chains)) {
+      roleKeysPerChain[chain as ChainName] = {
+        [Role.Relayer]: [],
+        [Role.Kathy]: [],
+      };
+      for (const role of rolesToFund) {
+        roleKeysPerChain[chain][role] = [
+          new LocalAgentKey(
+            environment,
+            role,
+            fundableRoleKeys[role as FundableRole],
+            chain,
+          ),
+        ];
+      }
+    }
```

Comment:
this is slightly hard to read
consider combining these loops?

Reply:
done